### PR TITLE
DEV: Migrate publish page modal to Glimmer and DModal

### PR DESCRIPTION
**Pull request by Drenmi**
### What is this change?

This PR migrates the publish page modal to a Glimmer component and DModal.

### Notes

- Most of the code is lift-and-shift. However, the component state getters were implemented using meta-programming in the original controller. They have all been inlined here for clarity, searchability, etc.
- The milestone commits have been kept for easier review. Squashing to be done before merge.

### Verification

Tests still pass, and manual verification is here:

publish-page-modal

### CODE REVIEW

**Comment by CvX** [comment]
Looking at the gif - there seems to be a double footer/border?

File: app/assets/javascripts/discourse/app/components/modal/publish-page.hbs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+              <span class="invalid-slug">{{this.reason}}.</span>
+            {{/if}}
+          {{/if}}
+
+        </div>
+      </ConditionalLoadingSpinner>
+    {{/if}}
+  </:body>
+  <:footer>
+    <div class="modal-footer">
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/app/components/modal/publish-page.hbs:74]: I think you get this when using `:footer` already:

```
{{#if (has-block "footer")}}
  <div class="modal-footer">
    {{yield to="footer"}}
  </div>
{{/if}}
```

Comment (jjaffeux) [reply app/assets/javascripts/discourse/app/components/modal/publish-page.hbs:74]: I just saw @CvX's comment, this is the reason

Comment (Drenmi) [reply app/assets/javascripts/discourse/app/components/modal/publish-page.hbs]: Sharp eyes. 👀 Fixed now.

**Review by jjaffeux — APPROVED** [review]
LGTM

If you had the time I would recommend looking at using @glimmer/component instead of @ember/component, it's a good way to learn about the differences but up to you 👍

**Comment by Drenmi** [comment]
> If you had the time I would recommend looking at using @glimmer/component instead of @ember/component, it's a good way to learn about the differences but up to you 👍

Hm. Not sure if I'm missing something. I am using `@glimmer/component` for the `publish-page` component [here](https://github.com/discourse/discourse/pull/22663/files#diff-562c05a65aef3ff5f0d39b62455766a1209867f1882b5e5741e799af0ed90ff1R4). 🤔

**Comment by jjaffeux** [comment]
> Hm. Not sure if I'm missing something. I am using `@glimmer/component` for the `publish-page` component [here](https://github.com/discourse/discourse/pull/22663/files#diff-562c05a65aef3ff5f0d39b62455766a1209867f1882b5e5741e799af0ed90ff1R4). 🤔

Yes my bad indeed, for some reason I have read "@ember/component" great thanks

**Pull request merged**